components/properties/src/script.rs

```diff
@@ -140,6 +171,72 @@ impl<'data> ScriptExtensions<'data> {
             ZeroSlice::from_ule_slice(script_ule_slice)
         }
     }
+
+    /// Returns whether `script` is contained in the Script_Extensions
+    /// property value if the code_point has Script_Extensions, otherwise
+    /// if the code point does not have Script_Extensions then returns
+    /// whether the Script property value matches.
+    ///
+    /// Some characters are commonly used in multiple scripts. For more information,
+    /// see UAX #24: http://www.unicode.org/reports/tr24/.
+    pub fn has_script(&self, code_point: u32, script: Script) -> bool {
+        let sc_with_ext = self.trie.get(code_point);
+
+        if !sc_with_ext.has_extensions() {
+            let script_val = sc_with_ext.0 & SCRIPT_X_SCRIPT_VAL;
+            script == Script(script_val)
+        } else {
+            let scx_val = self.get_script_extensions_val(code_point);
+            let script_find = scx_val.iter().find(|&sc| sc == script);
+            script_find.is_some()
+        }
+    }
+
+    /// Returns a [`UnicodeSet`] for the given [`Script`] which represents all
+    /// code points for which `has_script` will return true.
+    pub fn get_script_extensions_set(&self, script: Script) -> UnicodeSet {
+        let mut builder = UnicodeSetBuilder::new();
+        let mut range_start: u32 = 0;
+
+        loop {
+            let next_cpm_range = self.trie.get_range(range_start);
+
+            match next_cpm_range {
+                Some(cpm_range) => {
+                    let end = cpm_range.get_end();
+                    if end <= CODE_POINT_MAX {
+                        let start = cpm_range.get_start();
+                        let sc_with_ext = ScriptWithExt(cpm_range.get_value() as u16);
+
+                        // Since we already have the ScriptWithExt value from the trie,
+                        // we can directly check if `script` matches to code point `start`
+                        // without calling `has_script()`
+                        if (!sc_with_ext.has_extensions() && script == sc_with_ext.into())
+                            || (sc_with_ext.has_extensions()
+                                && self
+                                    .get_script_extensions_val(start)
+                                    .iter()
+                                    .any(|sc| sc == script))
+                        {
+                            builder.add_range_u32(&(start..=end));
```

Comment:
Suggestion: If you make `CPT::iter_ranges()` return a Rust Iterator, then you can make this a `filter` and return out another iterator without realizing it into an actual UnicodeSet.  You can then make a `From` impl on UnicodeSet that allows the CPTRangeIterator to be converted into a UnicodeSet if the client wants it.

Reply:
> As far as the From impl, it would create a dependency on CPT from UnicodeSet.

The impl can go into either crate.  I imagine CPT depends on UnicodeSet, so the impl should go into the CPT crate.

Reply:
Done. 

Also, even though `CPT:iter_ranges()` returns an iterator of `CodePointMapRange`, after the decision to change the return type of `get_script_extension_ranges` to only return `RangeInclusive`, we don't need CPT to depend on UnicodeSet for that particular case. I was able to put the conversion in `icu_uniset` without adding dependencies.